benchmark/single-source/DictionaryRemove.swift

```diff
@@ -26,6 +26,12 @@ public let benchmarks = [
     runFunction: remove, tags: t, legacyFactor: 10),
   BenchmarkInfo(name: "DictionaryRemoveOfObjects",
     runFunction: removeObjects, tags: t, legacyFactor: 100),
+
+  BenchmarkInfo(name: "DictionaryFilter",
+    runFunction: filter, tags: t, legacyFactor: 10),
```

Comment:
The legacy factor is only needed for older benchmark functions that we (used to?) not be able to change without breaking our ability to compare benchmark results across compiler releases. We should not use it for new definitions -- instead, I recommend just incorporating this into the `100*n` factors below. (iirc, changing it to `1000*n` should work. (Or is it `10*n`?))